mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -267,7 +267,8 @@ private boolean isInAttachments(String value, List<MessageAttachmentMetadata> at
             return textExtractor
                     .extractContent(
                         rawData,
-                        attachment.getType())
+                        // todo we likely want only the media type here
+                        attachment.getType().asString())
```

Comment:
why? a text extractor should not be aware of the encoding?

Reply:
It's better if you tell it, rather than just "guess it".

Tika is able to guess the encoding, at the price of a performance hit.

JSoup, default text extractor... Not sure they can do such a thing...